capstone-frontend/src/components/CreateList.jsx

```diff
@@ -1,28 +1,99 @@
 import React, { Component } from 'react';
 import { withRouter } from 'react-router-dom'
-import { Button, Form, Spinner, Modal } from 'react-bootstrap'
+import { Button, Form, Spinner, Modal, Col, Row } from 'react-bootstrap'
 
 class CreateList extends Component {
 
   constructor(props) {
     super(props)
 
     this.state = {
-      loading: false,
+      creatingBookList: false, // For Spinner
+      fetchingBooks: false, // For Spinner
       showModal: false,
+      typingTimeout: 0,
+      searchTerm: "",
+      searchResults: [],
+      displayBooks: false,
+      addedBooksIDs: [],
+      addedBooks: [],
     }
   }
 
-  handleSubmit = async () => {
+  getBooks = async (searchTerm) => {
+
+    this.setState({ fetchingBooks: true })
+
+    let searchResults;
+
+    if (searchTerm === "") {
+      searchResults = [];
+      
+      this.setState({ searchResults, displayBooks: false, fetchingBooks: false })
+    }
+    else {
+      searchResults = await fetch(`/api/search?searchTerm=${searchTerm}&maxResults=${4}`)
```

Comment:
nit: consider storing magic numbers like the `maxResults` parameter and the `typingTimeout` below as constants.